### PR TITLE
Add the ability to customize the generated provider name 

### PR DESCRIPTION
**Pull request by mansourzaki**
Hello!
I tried to add the ability to customize the generated provider name for this issue #2244

### CODE REVIEW

File: packages/riverpod_analyzer_utils/lib/src/riverpod_element.dart [hunk 1]
```diff
@@ -11,6 +11,7 @@ class RiverpodAnnotationElement {
   @internal
   RiverpodAnnotationElement({
```
Comment (rrousselGit) [inline packages/riverpod_analyzer_utils/lib/src/riverpod_element.dart:12]: Changes in this file needs to be tested. Tests are located in a different folder within `packages` with a similar name,

Comment (rrousselGit) [reply packages/riverpod_analyzer_utils/lib/src/riverpod_element.dart:12]: Generator changes also need to be tested.


And you'd also need to test that this new feature works well with riverpod_lint (specifically for lints like `provider_dependencies`)

File: packages/riverpod_annotation/lib/src/riverpod_annotation.dart [hunk 2]
```diff
@@ -23,6 +24,11 @@ class Riverpod {
   /// Defaults to false.
   final bool keepAlive;
 
+  /// To set a custom name for the provider you want to generate.
+  ///
+  /// Defaults to null.
+  final String? name;
```
Comment (bleonard252) [inline packages/riverpod_annotation/lib/src/riverpod_annotation.dart:30]:
```suggestion
  /// A custom name for the provider you want to generate.
  /// Riverpod adds a suffix to this, which defaults to `Provider`. Set the `provider_name_suffix` option in build_runner settings to change this; see [riverpod_generator](https://pub.dev/packages/riverpod_generator#global-configuration) for more details.
  ///
  /// Defaults to the name of the annotated method.
  final String? name;
```

The documentation for this is a bit unclear.

**Comment by rrousselGit** [comment]
Flagging as draft since this still needs testing :)

**Comment by rrousselGit** [comment]
:wave:

Since this seems stale, I'll close this PR. Feel free to reopen a PR if you can resume working on this :)

**Pull request closed without merging**